pages/api/checkIn.ts

```diff
@@ -146,12 +145,7 @@ function verifyCorrectLocation(
   const lastLocation = locations.find(
     (location) => location.key.toString() === userState.lastLocation.toString()
   )
-
-  if (!lastLocation || currentLocation.index !== lastLocation.index + 1) {
-    return false
-  } else {
-    return true
-  }
+  return !(!lastLocation || currentLocation.index !== lastLocation.index + 1)
```

Comment:
This code is hard to read, but so was the previous code. Per https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#code

> You're writing code to be read, understood, and changed by others.

Let's emphasize understanding over terseness.

It returns a boolean, so I'd call it `checkCorrectLocation()`

Then at the end, I'd use a variable name for whatever `currentLocation.index !== lastLocation.index + 1` is supposed to represent. 

```
if ( ! lastLocation ) {
  return false
}

const locationHasChanged =  currentLocation.index !== lastLocation.index + 1
return ! locationHasChanged
```

You're welcome to use better variable names.

Reply:
Thanks for the review @mikemaccana , it is updated.